packages/block-editor/src/components/block-inspector/index.js

```diff
@@ -171,6 +176,49 @@ const BlockInspector = ( { showNoBlockSelectedMessage = true } ) => {
 	const availableTabs = useInspectorControlsTabs( blockType?.name );
 	const showTabs = availableTabs.length > 1;
 
+	const isOffCanvasNavigationEditorEnabled =
+		window?.__experimentalEnableOffCanvasNavigationEditor === true;
+
+	const isNavBlock = blockType.name === 'core/navigation' ? true : false;
+
+	const { parentNavBlockClientId } = useSelect( ( select ) => {
+		const { getSelectedBlockClientId, getBlockParentsByBlockName } =
+			select( blockEditorStore );
+
+		const _selectedBlockClientId = getSelectedBlockClientId();
+
+		return {
+			parentNavBlockClientId: getBlockParentsByBlockName(
+				_selectedBlockClientId,
+				'core/navigation',
+				true
+			)[ 0 ],
+		};
+	}, [] );
```

Comment:
In https://github.com/WordPress/gutenberg/pull/46058 I tried to make the whole "Back button" thing universal with the only exceptions being the Template Part and Reusable Blocks.

If we were to persue that route we could remove this code and make it generic to the "nearest controlling parent" right?

Reply:
Another option might be to investgate whether we can configure the animation based on information passed via block editor settings.

Here's some [prior art](https://github.com/WordPress/gutenberg/pull/46321/files) on that one.

So you could have a setting which determines the specific blocks that will exhibit the special "back button" behaviour. 

This is all to avoid coupling a block to the block-editor package.